src/iuse_actor.cpp

```diff
@@ -1863,3 +1863,362 @@ long holster_actor::use( player *p, item *it, bool, const tripoint & ) const
 
     return 0;
 }
+
+void repair_item_actor::load( JsonObject &obj )
+{
+    // Mandatory:
+    JsonArray jarr = obj.get_array( "materials" );
+    while( jarr.has_more() ) {
+        materials.push_back( jarr.next_string() );
+    }
+
+    // TODO: Make skill non-mandatory while still erroring on invalid skill
+    const std::string skill_string = obj.get_string( "skill" );
+    used_skill = skill_id( skill_string );
+    if( !used_skill.is_valid() ) {
+        throw JsonError( std::string( "Invalid skill for repair_item_actor: " ) + skill_string );
```

Comment:
You should probably call `obj.throw_error` instead of throwing here. The function adds the surrounding JSON context to the error message. It's second parameter is the member name that caused the error, so it should look like this:
```C++
obj.throw_error("Invalid skill", "skill");
```
(Technically, the skill id does not need to be part of the error message as it's clear from the surrounding JSON context, which is determined by the member name.)